symengine/tests/basic/test_polynomial.cpp

```diff
@@ -25,21 +29,167 @@ using SymEngine::integer_class;
+    RCP<const UnivariateIntPolynomial> po = univariate_int_polynomial(x, 5, {{5, 1}});
+    RCP<const UnivariateIntPolynomial> poly = univariate_int_polynomial(x, 2, {{0, 1}, {1, 2}, {2, 1}});
+
+    REQUIRE((z->is_zero() and not z->is_one() and not z->is_minus_one() and z->is_integer() and not z->is_symbol() and
+			not z->is_mul() and not z->is_pow()));
+    REQUIRE((not o->is_zero() and o->is_one() and not o->is_minus_one() and o->is_integer() and not o->is_symbol() and
+			not o->is_mul() and not o->is_pow()));
+    REQUIRE((not mo->is_zero() and not mo->is_one() and mo->is_minus_one() and mo->is_integer() and not mo->is_symbol() and
+			not mo->is_mul() and not mo->is_pow()));
+    REQUIRE((not i->is_zero() and not i->is_one() and not i->is_minus_one() and i->is_integer() and not i->is_symbol() and
+			not i->is_mul() and not i->is_pow()));
+    REQUIRE((not s->is_zero() and not s->is_one() and not s->is_minus_one() and not s->is_integer() and s->is_symbol() and
+			not s->is_mul() and not s->is_pow()));
+    REQUIRE((not m1->is_zero() and not m1->is_one() and not m1->is_minus_one() and not m1->is_integer() and not m1->is_symbol() and
+			m1->is_mul() and not m1->is_pow()));
+    REQUIRE((not m2->is_zero() and not m2->is_one() and not m2->is_minus_one() and not m2->is_integer() and not m2->is_symbol() and
+			m2->is_mul() and not m2->is_pow()));
+    REQUIRE((not po->is_zero() and not po->is_one() and not po->is_minus_one() and not po->is_integer() and not po->is_symbol() and
+			not po->is_mul() and po->is_pow()));
+    REQUIRE((not poly->is_zero() and not poly->is_one() and not poly->is_minus_one() and not poly->is_integer() and not poly->is_symbol() and
+			not poly->is_mul() and not poly->is_pow()));
+}
+
+TEST_CASE("Univariate Int Polynomial expand", "[UnivariateIntPolynomial][expand]")
+{
+    RCP<const Symbol> x  = symbol("x");
+    RCP<const UnivariateIntPolynomial> a = univariate_int_polynomial(x, 3, {{1, 1}, {2, 1}, {3, 1}});
+    RCP<const Basic> b = make_rcp<const Pow>(a, integer(3));
+    RCP<const Basic> c = expand(b);
+
+//    std::cout<<b->__str__()<<std::endl;
+    REQUIRE(b->__str__() == "(x**3 + x**2 + x)**3");
+    REQUIRE(c->__str__() == "x**9 + 3*x**8 + 6*x**7 + 7*x**6 + 6*x**5 + 3*x**4 + x**3");
+    //std::cout<<c->__str__()<<std::endl;
+}
+
+TEST_CASE("Constructor of UnivariatePolynomial", "[UnivariatePolynomial]")
+{
+    RCP<const Symbol> x  = symbol("x");
+    Expression a(symbol("a"));
+    Expression b(integer(2));
+    Expression c(integer(1));
+    RCP<const UnivariatePolynomial> P = univariate_polynomial(x, 2, {{0, c}, {1, b}, {2, a}});
+    REQUIRE(P->__str__() == "a*x**2 + 2*x + 1");
+
+    RCP<const UnivariatePolynomial> Q = UnivariatePolynomial::create(x, {1, 0, 2, 1});
+    REQUIRE(Q->__str__() == "x**3 + 2*x**2 + 1");
+
+    RCP<const UnivariatePolynomial> R = univariate_polynomial(x, 2, {{0, c}, {1, b}, {2, a}, {3, c}});
```

Comment:
Here, value `2` is wrong. That's why there is a not canonical error here,
`
SYMENGINE_ASSERT failed: /Users/irislui/Documents/ECS_193/SymEngine/symengine/symengine/polynomial.cpp
function UnivariatePolynomial(), line number 246 at 
is_canonical(degree_, dict_)
`